src/components/text-input/Label.root.tsx

```diff
@@ -0,0 +1,60 @@
+import styled, { css } from 'styled-components';
+import { FlattenSimpleInterpolation, ThemedStyledProps } from 'styled-components/ts3.6';
+
+import { applyTextStyles } from '../text';
+import { Utils } from '../../shared';
+import { applyMargin, applyPadding } from '../../utilities/layout';
+import { TTheme } from '../../utilities/theme';
+import Spacing from '../../utilities/spacing';
+
+import { TEXT_INPUT_VALUES_MAPPING, LABEL_POSITIONS } from './TextInput.constants';
+import { PLabelRoot } from './TextInput.props';
+
+const LabelRoot = styled.label<PLabelRoot>(
+    ({
+        theme: { background, animation },
+        size,
+        value,
+        animatedLabel,
+        leadingIcon,
+        backgroundColor,
+    }: ThemedStyledProps<PLabelRoot, TTheme>): FlattenSimpleInterpolation => {
+        const hasValue = Utils.isString(value) && value.length > 0;
+
+        const colors: Record<string, string> = {
+            background: backgroundColor || background.default,
+        };
```

Comment:
this is not needed, since we do not re-assign any values ... you can directly add this to the outputted css